Redis

valentinyanakiev opened the pull request.

valentinyanakiev committed:
Redis

valentinyanakiev committed:
Merge branch 'develop' into redis

valentinyanakiev committed:
Cleanup + mocks

valentinyanakiev committed:
Replaced locala CacheModule references with global one

valentinyanakiev committed:
Fixed User + UserGroup tests

valentinyanakiev committed:
Fixed Challenge tests

valentinyanakiev committed:
Fixed platform tests

valentinyanakiev committed:
Fixed community tests

valentinyanakiev committed:
Collaboration tests

valentinyanakiev committed:
Common tests

valentinyanakiev committed:
Refactored mocker factory

valentinyanakiev committed:
Add / fix comms tests

valentinyanakiev committed:
Fixed context tests

valentinyanakiev committed:
Wrapped up mocking all src/domain tests suits

valentinyanakiev committed:
Mocked all src/services services in tests

valentinyanakiev committed:
Mocked all src/core tests

valentinyanakiev committed:
Remove travis dependency

valentinyanakiev committed:
Run unit tests in parallel

hero101 commented on src/domain/communication/communication/communication.service.spec.ts:
```diff
@@ -1,14 +1,29 @@
-    }).compile();
+      providers: [
+        CommunicationService,
+        {
```
extract this provider and replace it inline like the others

valentinyanakiev replied:
Fixed with a factory in https://github.com/alkem-io/server/pull/1997/commits/5433ab6f5485f5d2000b8369104266935623c945

hero101 reviewed (changes requested):
Suggestions aside, I am not sure you need to use `useMocker` if you are providing the dependencies mocked in each test.
Also, take a look at [this](https://jestjs.io/docs/configuration#automock-boolean).

valentinyanakiev committed:
pubSubEngineModuleMockFactory --> pubSubEngineMockFactory

pubSubEngineModuleMockFactory --> pubSubEngineMockFactory

pubSubEngineModuleMockFactory --> pubSubEngineMockFactory

valentinyanakiev committed:
PR suggestions

valentinyanakiev committed:
Global cache module
